app/Http/Controllers/PageController.php

```diff
@@ -139,8 +140,12 @@ public function api_show(Page $page)
      */
     public function edit($id)
     {
+        if(!Gate::allows('web')){
+            abort(404, "Access denied");
+        }
         $page = Page::find($id);
         $images = Image::All();
+        $components = Component::All();
```

Comment:
Remove this line. We only want parent components, not all of them.

Reply:
Removed 👍